frontend/src/components/widgets/DataFrame/columns/ImageColumn.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GridCell, GridCellKind, ImageCell } from "@glideapps/glide-data-grid"
+
+import { notNullOrUndefined } from "src/lib/utils"
+
+import { BaseColumn, BaseColumnProps, toSafeString } from "./utils"
+
+/**
+ * A column type that renders an image as a cell value. On cell selection, open
+ * the cell overlay with a full version of the image.
+ *
+ * This column type is currently read-only.
+ */
+function ImageColumn(props: BaseColumnProps): BaseColumn {
+  const cellTemplate = {
+    kind: GridCellKind.Image,
+    data: [],
+    displayData: [],
+    allowAdd: false,
+    allowOverlay: true,
+    contentAlign: props.contentAlignment || "center",
+    style: props.isIndex ? "faded" : "normal",
+  } as ImageCell
+
+  return {
+    ...props,
+    kind: "image",
+    sortMode: "default",
+    isEditable: false, // Image columns are always read-only
+    getCell(data?: any): GridCell {
+      const imageUrls = notNullOrUndefined(data) ? [toSafeString(data)] : []
```

Comment:
Hm why is this an array out of curiosity?

Reply:
The native image cell of glide-data-grid is implemented to support an array of images. These are displayed in a gallery. That's why we implement our own single-image overlay.

Reply:
I added a comment